### ISSUE TITLE
How to profile

### ISSUE THREAD

**Issue by YLSnowy** [LLMServe/DistServe#13]
I want to get the profile data in my own experimental environment. How can I do it?
It seems to be able to do this with this command: "python distserve/example/profile.py --model facebook/opt-1.3b --tokenizer facebook/opt-1.3b --beam_width 1 --file_path <file_path>"
But I didn't find profile.py or similar code.

**Comment by interestingLSY** [comment]
Just ignore the profiling file. It is not used in the current scheduler.

**Comment by YLSnowy** [comment]
Oh, I mean I want to get my profile data. It is used in all the scripts in "ae-scripts/e2e" or "evaluation/e-benchmark/serving"

**Comment by YLSnowy** [comment]
> Just ignore the profiling file. It is not used in the current scheduler.
Oh, I mean I want to get my profile data. It is used in all the scripts in "ae-scripts/e2e" or "evaluation/e-benchmark/serving"

**Comment by PKUFlyingPig** [comment]
It is only needed when using the simulator to find the optimal placement strategy. If you only have limited number of GPUs, you can manually set the parallelism strategy by yourself.

**Comment by interestingLSY** [comment]
> Oh, I mean I want to get my profile data. It is used in all the scripts in "ae-scripts/e2e" or "evaluation/e-benchmark/serving"

May you please point out where we have used that file? I remember that the profiling file is not used in the project, nor can I find it.

**Comment by YLSnowy** [comment]
> It is only needed when using the simulator to find the optimal placement strategy. If you only have limited number of GPUs, you can manually set the parallelism strategy by yourself.

Yeah. I want to quickly find the best parallel strategy in my experimental environment for e2e latency. I hope to use the simulator instead of the exhaustive method. However, some of the parallel strategies I want to try are not in the profiling data, so can I ask how you profiled them?

**Comment by PKUFlyingPig** [comment]
What is the cluster setting? e.g., #nodes, #gpus, GPU type, intra-/inter-node network bandwidth?

**Comment by PKUFlyingPig** [comment]
The profiling process is quite straightforward, you profile the forward-pass time of the model under a specific parallelism strategy to get sufficient data to fit the latency model as proposed in Appendix A of our paper. Then you can use our simulator to find the optimal placement.

**Comment by YLSnowy** [comment]
Thank you
